docs/Problems.md

```diff
@@ -1,9 +1,123 @@
 # Common Problems
 
-## Sending without Reading
-
 ## Blocking the World
 
+```python
+import socket
+import idiokit
+from idiokit.http.server import serve_http
+
+
+@idiokit.stream
+def handler(addr, request, response):
+    _, _, query = request.uri.partition("/")
+    try:
+        results = socket.getaddrinfo(query, None, socket.AF_INET)
+    except socket.gaierror:
+        pass
+    else:
+        ipv4_addresses = set(result[4][0] for result in results)
+        yield response.write("\n".join(ipv4_addresses) + "\n")
+
+
+idiokit.main_loop(serve_http(handler, "localhost", 8080))
+```
+
+Use `idiokit.thread` or better yet, an asynchronous idiokit counterpart when available: In this case `idiokit.dns.a` does the trick.
+
+
+## Blocking the Pipeline: `idiokit.send` without `idiokit.next`
+
+```python
+import idiokit
+
+
+@idiokit.stream
+def produce():
+    for number in range(1000):
+        print "Sending out", number
+        yield idiokit.send(number)
+        yield idiokit.sleep(1.0)
+
+
+idiokit.main_loop(produce() | idiokit.consume())
```

Comment:
I suppose this line was supposed to be just

idiokit.main_loop(produce())

Because otherwise it will not idle forever (as explained in the solution below)?